opts/opts.go

```diff
@@ -54,6 +58,31 @@ type ListOpts struct {
 	validator ValidatorFctType
 }
 
+//MapOpts type
+type MapOpts struct {
+	values *map[string]string
+}
+
+func (opts *MapOpts) Set(value string) error {
+	vals := strings.Split(value, "=")
```

Comment:
That would allow one to specify an option like `--map-opt key=value1=value2` I cant think of a use case for something like that. Are there any potential benefits of doing that?